toolchain/driver/driver.cpp

```diff
@@ -830,4 +901,59 @@ auto Driver::Compile(const CompileOptions& options) -> RunResult {
   return make_result();
 }
 
+static void AddOSFlags(llvm::StringRef target,
+                       llvm::SmallVectorImpl<llvm::StringRef>& args) {
+  llvm::Triple triple(target);
+  switch (triple.getOS()) {
+    case llvm::Triple::Darwin:
+    case llvm::Triple::MacOSX:
+      // On macOS we need to set the sysroot to a viable SDK. Currently, this
+      // hard codes the path to be the unversioned symlink. The prefix is also
+      // hard coded in Homebrew and so this seems likely to work reasonably
+      // well. Homebrew and I suspect the Xcode Clang both have this hard coded
+      // at build time, so this seems reasonably safe but we can revisit if/when
+      // needed.
+      args.push_back(
+          "--sysroot=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk");
+      // We also need to insist on a modern linker, otherwise the driver tries
+      // to old and deprecated flags. The specific number here comes from an
+      // inspection of the Clang driver source code to understand where features
+      // were enabled, and this appears to be the latest version to control
+      // driver behavior.
+      //
+      // TODO: We should replace this with use of `lld` eventually.
+      args.push_back("-mlinker-version=705");
```

Comment:
Is it worth validating that these aren't already in args?

Reply:
My mental model for all the arg manipulation is basically around appending-to-override. And so the idea is that this will override any earlier versions, but we can allow further appending to override these back again.

I don't think the first will come up, so I can check that these don't exist yet, but it'll add some expense at minimal benefit.

I expect us to potentially want to allow overriding of these with later flags though. This is essentially setting up the defaults that users might override.

It's not in this PR, but thinking of exposing something like `--clang-option=-xyz` that lets us pass arbitrary extra flags to Clang long term.